### PR TITLE
Migrate connector to Ballerina lang vSLAlpha 4

### PR DESCRIPTION
**Pull request by aneeshafedo**
## Automation tests
 - Unit tests 
   > Done for all the operations 
 - Integration tests
   > Done for all the operations 

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: sfdc/auth_handler.bal [hunk 1]
```diff
@@ -1,53 +1,41 @@
...
+// //
+// // Unless required by applicable law or agreed to in writing,
+// // software distributed under the License is distributed on an
+// // "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// // KIND, either express or implied.  See the License for the
+// // specific language governing permissions and limitations
+// // under the License.
+// //
+// import ballerina/oauth2;
+// import ballerina/http;
```
Comment (indikasampath2000) [inline sfdc/auth_handler.bal:18]: It seems something not correct here. Please check.

File: sfdc/auth_handler.bal [hunk 2]
```diff
@@ -1,53 +1,41 @@
...
-# + authProvider - The `ClientOAuth2Provider` instance to handle auth requests. 
-public client class SalesforceAuthHandler {
+// # Representation of the Bearer Auth header handler for both inbound and outbound HTTP traffic.
+// #
+// # + authProvider - The `ClientOAuth2Provider` instance to handle auth requests. 
+// public client class SalesforceAuthHandler {
 
-    oauth2:ClientOAuth2Provider? authProvider;
-    string accessToken = "";
+//     oauth2:ClientOAuth2Provider authProvider;
```
Comment (indikasampath2000) [inline sfdc/auth_handler.bal:25]: Please remove unwanted code rather commenting them.

File: sfdc/auth_handler.bal [hunk 3]
```diff
@@ -1,53 +1,41 @@
...
+//         string|oauth2:Error token = self.authProvider.generateToken();
+//         if (token is string) {
+//             //req.setHeader(AUTHORIZATION, BEARER + token);
+//             req.setHeader(X_SFDC_SESSION, token);
+//             return req;
+//         } else {
+//             return prepareClientAuthError("Failed to enrich request with OAuth2 token.", token);
+//         }
+//     }
+// }
```
Comment (abeykoon) [inline sfdc/auth_handler.bal:41]: Please remove commented out code block.

File: sfdc/auth_handler.bal (hunk 1 above)
Comment (aneeshafedo) [reply sfdc/auth_handler.bal]: File deleted

File: sfdc/auth_handler.bal (hunk 2 above)
Comment (aneeshafedo) [reply sfdc/auth_handler.bal]: File deleted

File: sfdc/tests/bulk_csv_delete.bal [hunk 4]
```diff
@@ -46,41 +57,79 @@ function deleteCsv() {
...
+                if i != 5 {
+                    log:printWarn("getBatchResult Operation Failed! Retrying...");
+                    runtime:sleep(delayInSecs);
+                } else {
+                    log:printWarn("getBatchResult Operation Failed! Giving up...");
+                    test:assertFail(msg = batchResult.message());
+                }
+            } else {
+                test:assertFail("Invalid Batch Result!");
+            }
```
Comment (abeykoon) [inline sfdc/tests/bulk_csv_delete.bal:129]: Great! We have retry functionality in order now.

**Review by abeykoon — APPROVED** [review]

**Pull request merged**